streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -476,6 +477,19 @@ public void shouldNotGetTaskWithKeyAndPartitionerWhenNotRunning() {
         globalStreams.metadataForKey("store", "key", (topic, key, value, numPartitions) -> 0);
     }
 
+    class MyStringSerde<K> extends Serdes.WrapperSerde<String> {
```

Comment:
Should the superclass take `K` not `String` as type to not bind it to `String` type?

Also, we should write a new test method but not add it into an exiting one.

Reply:
I used current test - otherwise I would be copying existing test code

Reply:
One test should test only one thing IMHO. To share code, maybe refactor one private method that two public`@Test` methods can both use?

However, we should test not only `Consumed` but also `Produced`, `Grouped`, `Joined`, `Materialized` (maybe even deprecated `Produced`?). Thus, sharing code seems to be hard anyway.